tasks.py

```diff
@@ -1,36 +1,64 @@
+import shlex
 import subprocess
+import sys
 from pathlib import Path
+from typing import Optional
 
 import invoke
-from invoke import task
+from invoke import call, task
 
-CURDIR: Path = Path(__file__).absolute().parent
+ROOT: Path = Path(__file__).absolute().parent
+DOCS_IGNORE = ["devutils"]
 
 
 @task
-def poetry_lock(ctx: invoke.Context) -> None:
-    """Runs 'poetry lock' on all projects with a pyproject.toml."""
+def install(
+    ctx: invoke.Context,
+    update: bool = False,
+    verbose: bool = False,
+    skip: Optional[list[str]] = None,
+) -> None:
+    """Refresh/Install all Poetry-based projects' environments.
+
+    Args:
+        update: Whether to update the dependencies or just installing them.
+        verbose: Whether to run in verbose mode.
+        skip: List of project directories to skip.
+
+    Note:
+        Can't use threads to speed it up. (Poetry or pip will just fail)
+    """
     for project_dir in _iter_project_dirs():
-        # Note: can't use threads to speed it up. poetry (or pip) just fail...
-        subprocess.check_call(["poetry", "lock"], cwd=project_dir)
+        if skip and project_dir in skip:
+            print(f"Skipping project {project_dir!r}.")
+            continue
 
+        inv_cmd, poetry_cmd = "invoke install", "poetry install"
+        if update:
+            inv_cmd += " -u"
+            poetry_cmd = poetry_cmd.replace("install", "update")
+        if verbose:
+            inv_cmd += " -v"
+            poetry_cmd += " -vv"
 
-@task
+        cmd = inv_cmd if (project_dir / "tasks.py").exists() else poetry_cmd
+        subprocess.check_call(shlex.split(cmd), cwd=project_dir)
+
+
+@task(pre=[call(install, skip=DOCS_IGNORE)])
 def docs(ctx: invoke.Context) -> None:
     """Regenerate documentation for each library."""
-    ignored = [
-        "devutils",
-        "integration_tests",
-        "meta",
-    ]
+    ignored = DOCS_IGNORE[:]
+    if sys.platform != "win32":
+        ignored.append("windows")
```

Comment:
This won't be required once we enable **windows** dev on Mac/Linux for simple tasks, like docs generation. #264